receiver/awscontainerinsightreceiver/internal/neuron/neuron_monitor_scraper_test.go

```diff
@@ -0,0 +1,393 @@
+
+	scopeMetrics := md.ResourceMetrics().At(0).ScopeMetrics().At(0).Metrics()
+	for i := 0; i < scopeMetrics.Len(); i++ {
+		metric := scopeMetrics.At(i)
+		if metric.Name() == "neuroncore_utilization_ratio" {
+			assert.Equal(m.t, float64(0.1), metric.Gauge().DataPoints().At(0).DoubleValue())
+			*m.coreUtil = true
+		}
+
+		if strings.Contains(metric.Name(), "neuroncore") {
+			instanceId, _ := metric.Gauge().DataPoints().At(0).Attributes().Get("InstanceId")
+			podName, _ := metric.Gauge().DataPoints().At(0).Attributes().Get("PodName")
+			namespace, _ := metric.Gauge().DataPoints().At(0).Attributes().Get("Namespace")
+			containerName, _ := metric.Gauge().DataPoints().At(0).Attributes().Get("Namespace")
+			_, relabeled := metric.Gauge().DataPoints().At(0).Attributes().Get("DeviceId")
+			*m.instanceId = instanceId.Str() == "i-09db9b55e0095612f"
+			*m.podName = podName.Str() == nueronCoreID
+			*m.namespace = namespace.Str() == nueronCoreID
+			*m.containerName = containerName.Str() == nueronCoreID
+			*m.relabeled = relabeled
+		}
+
+		if strings.Contains(metric.Name(), "system") {
+			instanceId, _ := metric.Gauge().DataPoints().At(0).Attributes().Get("InstanceId")
+			_, podNameFound := metric.Gauge().DataPoints().At(0).Attributes().Get("PodName")
+			_, namespaceFound := metric.Gauge().DataPoints().At(0).Attributes().Get("Namespace")
+			_, relabelFound := metric.Gauge().DataPoints().At(0).Attributes().Get("DeviceId")
+			assert.Equal(m.t, instanceId.Str(), "i-09db9b55e0095612f")
+			assert.False(m.t, podNameFound)
+			assert.False(m.t, namespaceFound)
+			assert.False(m.t, relabelFound)
+		}
+
+		if metric.Name() == "neuron_runtime_memory_used_bytes" {
+			assert.Equal(m.t, float64(9.043968e+06), metric.Gauge().DataPoints().At(0).DoubleValue())
+			*m.memUsed = true
+		}
+
+		if metric.Name() == "up" {
+			assert.Equal(m.t, float64(1.0), metric.Gauge().DataPoints().At(0).DoubleValue())
+			*m.up = true
+		}
+	}
+
+	return nil
+}
+
+func TestNewNeuronScraperBadInputs(t *testing.T) {
+	settings := componenttest.NewNopTelemetrySettings()
+	settings.Logger, _ = zap.NewDevelopment()
+
+	tests := []NeuronMonitorScraperOpts{
+		{
+			Ctx:               context.TODO(),
+			TelemetrySettings: settings,
+			Consumer:          nil,
+			Host:              componenttest.NewNopHost(),
+			HostInfoProvider:  mockHostInfoProvider{},
+		},
+		{
+			Ctx:               context.TODO(),
+			TelemetrySettings: settings,
+			Consumer:          mockConsumer{},
+			Host:              nil,
+			HostInfoProvider:  mockHostInfoProvider{},
+		},
+		{
+			Ctx:               context.TODO(),
+			TelemetrySettings: settings,
+			Consumer:          mockConsumer{},
+			Host:              componenttest.NewNopHost(),
+			HostInfoProvider:  nil,
+		},
+	}
+
+	for _, tt := range tests {
+		scraper, err := NewNeuronMonitorScraper(tt)
+
+		assert.Error(t, err)
+		assert.Nil(t, scraper)
+	}
+}
+
+func TestNewNeuronScraperEndToEnd(t *testing.T) {
```

Comment:
There is a lot of duplication between this and the other Prometheus scraper tests, and it makes it hard to read the test when there's so much cruft.  Can we consolidate some?  I would want to be able to write a test like:

```
const neuronMonitorOutput = "..."

func TestEndToEnd(t *testing.T) {
    consumer := createNeuronConsumer()
    scraper := createScraper(neuron.scrapeConfig, neuronMonitorOutput)
    metrics := scraper.GetMetrics()
    // validate
}
```

I'm probably missing some stuff that you'd need to pass in, but I hope a lot of it can be abstracted away so the main test code is really readable.

Reply:
consolidated into one class